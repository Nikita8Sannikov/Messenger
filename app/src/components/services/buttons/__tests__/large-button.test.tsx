import { render, screen } from "@testing-library/react";

import LargeButtonComponent from "@components/services/buttons/large-button";

describe("LargeButtonComponent", () => {
	test("renders with children", () => {
		render(<LargeButtonComponent>Test Button</LargeButtonComponent>);
		const button = screen.getByText("Test Button");
		expect(button).toBeInTheDocument();
	});

	test("applies large size prop", () => {
		render(<LargeButtonComponent>Test Button</LargeButtonComponent>);
		const button = screen.getByText("Test Button");
		expect(button).toHaveClass("MuiButton-sizeLarge");
	});

	test("passes through other props correctly", () => {
		render(
			<LargeButtonComponent 
				variant="contained" 
				color="primary"
				disabled
				className="custom-class"
				fullWidth
			>
                Test Button
			</LargeButtonComponent>,
		);
		const button = screen.getByText("Test Button");
		expect(button).toHaveClass("MuiButton-contained");
		expect(button).toHaveClass("MuiButton-colorPrimary");
		expect(button).toHaveClass("custom-class");
		expect(button).toHaveClass("MuiButton-fullWidth");
		expect(button).toBeDisabled();
	});

	test("handles click events", () => {
		const handleClick = jest.fn();
		render(
			<LargeButtonComponent onClick={handleClick}>
                Test Button
			</LargeButtonComponent>,
		);
		const button = screen.getByText("Test Button");
		button.click();
		expect(handleClick).toHaveBeenCalledTimes(1);
	});

	test("matches snapshot", () => {
		const { asFragment } = render(
			<LargeButtonComponent variant="contained" color="primary" fullWidth>
                Test Button
			</LargeButtonComponent>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
