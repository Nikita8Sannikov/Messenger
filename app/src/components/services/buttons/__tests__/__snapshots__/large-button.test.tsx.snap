// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeButtonComponent matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-fullWidth css-13tpzue-MuiButtonBase-root-MuiButton-root"
    id=":r4:"
    tabindex="0"
    type="button"
  >
    Test Button
  </button>
</DocumentFragment>
`;
